BinaryEncoding.md

```diff
@@ -73,27 +73,37 @@ Strings referenced by the module (i.e. function names) are encoded as null-termi
 The following documents the current prototype format. This format is based on and supersedes the v8-native prototype format, originally in a [public design doc](https://docs.google.com/document/d/1-G11CnMA0My20KI9D7dBR6ZCPOBCRD0oCH6SHCPFGx0/edit?usp=sharing).
 
 ## High-level structure
-A module contains a sequence of sections, each identified by a *section identifier byte* whose encodings
-are listed below.
+A module contains a sequence of sections. Sections can in general be repeated, but some can occur only once or have dependant sections that must preceed them. Each section is identified by an immediate string. Sections whose identity is unknown to the WebAssembly implementation are ignored and this is supported by including the size in bytes for all sections. The encoding of all sections begins as follows:
+
+| Field | Type | Description |
+| ----- |  ----- | ----- |
+| id_len | `varuint32` | section identifier string length |
+| id_str | `char*` | section identifier string |
```

Comment:
In general we don't want `char*`, we instead want a leb128 size followed by a byte string (it doesn't even have to be valid Unicode).